modules/uploads/utils/bucket.ts

```diff
@@ -0,0 +1,286 @@
+			secretAccessKey: config.secretAccessKey,
+		},
+		endpoint: config.endpoint,
+		defaultUserAgentProvider: () =>
+			Promise.resolve([
+				["opengb/uploads", "0.1.0"],
+			]),
+	});
+}
+
+/**
+ * Create a presigned URL to perform a SINGLE PART upload to S3
+ * 
+ * @param config S3 Config
+ * @param uploadId The ID of the Upload batch
+ * @param file The file to upload
+ * @param expirySeconds How long the upload URL should be valid for
+ * @returns A URL that exposes an HTTP PUT endpoint to upload the file
+ */
+export async function getPresignedPutUrl(
+	config: S3Config,
+	uploadId: string,
+	file: UploadFile,
+	expirySeconds = 60 * 60,
+): Promise<{ key: string, urls: PresignedUrl[] }> {
+	const client = getClient(config);
+
+	const key = getKey(uploadId, file.path);
+	const command = new PutObjectCommand({
+		Bucket: config.bucket,
+		Key: key,
+		ContentType: file.mime ?? undefined,
+		ContentLength: parseInt(file.contentLength),
+	});
+	const url = await getSignedUrl(
+		client,
+		command,
+		{ expiresIn: expirySeconds },
+	);
+
+	return {
+		urls: [{
+			url,
+			partNumber: 0,
+			offset: "0",
+			contentLength: file.contentLength,
+		}],
+		key,
+	};
+}
+
+export async function getPresignedMultipartUploadUrls(
+	config: S3Config,
+	uploadId: string,
+	file: UploadFile,
+	chunkSize: bigint,
+	expirySeconds = 60 * 60 * 6,
+): Promise<{ key: string, urls: PresignedUrl[], multipartUploadId: string }> {
+	const client = getClient(config);
+
+	const key = getKey(uploadId, file.path);
+	const command = new CreateMultipartUploadCommand({
+		Bucket: config.bucket,
+		Key: key,
+		ContentType: file.mime ?? undefined,
+		
+	});
+	const multipartUpload = await client.send(command);
+
+	const id = multipartUpload.UploadId;
+	if (!id) throw new Error("Multipart upload ID not returned");
+
+	// Round up to the nearest chunk count
+	const chunkCount = (BigInt(file.contentLength) + chunkSize - 1n) / chunkSize;
+
+	const urls: PresignedUrl[] = [];
+	for (let i = 0n; i < chunkCount; i++) {
+		const offset = i * chunkSize;
+		const remaining = BigInt(file.contentLength) - offset;
+		const length = remaining < chunkSize ? remaining : chunkSize;
+		const partNumber = Number(i) + 1; // S3's part number is 1-based because reasons
+
+		const command = new UploadPartCommand({
+			Bucket: config.bucket,
+			Key: key,
+			UploadId: id,
+			ContentLength: Number(length),
+			PartNumber: partNumber, 
+		});
+
+		const chunkPutUrl = await getSignedUrl(
+			client,
+			command,
+			{ expiresIn: expirySeconds },
+		);
+
+		urls.push({
+			url: chunkPutUrl,
+			partNumber,
+			offset: offset.toString(),
+			contentLength: length.toString(),
+		});
+	}
+
+	return {
+		urls,
+		multipartUploadId: id,
+		key,
+	};
+}
+
+
+export async function getMultipartUploadParts(
+	config: S3Config,
+	key: string,
+	multipartUploadId: string,
+): Promise<{ PartNumber: number, ETag: string }[]> {
+	const client = getClient(config);
+
+	console.log(multipartUploadId);
```

Comment:
debug log?

Reply:
Left in by accident actually. Good catch!